### PR TITLE
fix as_strided_scatter_backward

### PR DESCRIPTION
**Pull request by bdhirsh**
as_strided_scatter's derivative formula was broken - instead of making a "mask" of 1's and 0's, it would effectively make a mask of 1's and uninitialized memory.

Fixes https://github.com/pytorch/pytorch/issues/88105

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88198
* #88230
* #88197
* #87647
* __->__ #87646
* #87255

cc @albanD

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Great catch!
Also I guess this means we don't have opinfo (and thus autograd testing) for these ops... We should add them!

**Comment by bdhirsh** [comment]
we had a few skipped's OpInfo's for `as_strided_scatter` that I just cleaned up. There are still a few more, but... the OpInfo for `as_strided()` has some skips as well.

**Review by albanD — APPROVED** [review]

File: test/functorch/test_aotdispatch.py [hunk 1]
```diff
@@ -952,7 +952,6 @@ def assert_compiler(gm: torch.fx.GraphModule, _):
     xfail('linalg.eig'),
     xfail('scatter_reduce', 'prod'),
 
-    # non-deterministic
```
Comment (albanD) [inline test/functorch/test_aotdispatch.py:955]: Why remove this?

**Review by albanD — APPROVED** [review]

File: test/functorch/test_aotdispatch.py (hunk 1 above)
Comment (bdhirsh) [reply test/functorch/test_aotdispatch.py:955]: I thought the non-determinism was caused by the uninitialized memory 😛

**Comment by mruberry** [comment]
If this goes in after https://github.com/pytorch/pytorch/pull/85583/files then it should review the skips on the OpInfo

File: test/functorch/test_aotdispatch.py (hunk 1 above)
Comment (mruberry) [reply test/functorch/test_aotdispatch.py:955]: Isn't the non-determinism that there may be multiple values mapping to the same index in the result, and which value appears there is nondeterministic? cc @kurtamohler

**Pull request closed without merging**